texar/torch/modules/tokenizers/pretrained_bert_tokenizer_utils.py

```diff
@@ -0,0 +1,299 @@
+        tokens = reader.readlines()
+    for index, token in enumerate(tokens):
+        token = token.rstrip('\n')
+        vocab[token] = index
+    return vocab
+
+
+class BasicTokenizer:
+    r"""Runs basic tokenization (punctuation splitting, lower casing, etc.).
+
+    Args:
+        do_lower_case: Whether to lower case the input.
+        never_split: (`optional`) list of str
+            Kept for backward compatibility purposes.
+            Now implemented directly at the base class level
+            List of token not to split.
+        tokenize_chinese_chars: (`optional`) boolean (default True)
+            Whether to tokenize Chinese characters.
+            This should likely be deactivated for Japanese:
+    """
+
+    def __init__(self, do_lower_case: bool = True,
+                 never_split: Optional[List[str]] = None,
+                 tokenize_chinese_chars: bool = True):
+        if never_split is None:
+            never_split = []
+        self.do_lower_case = do_lower_case
+        self.never_split = never_split
+        self.tokenize_chinese_chars = tokenize_chinese_chars
+
+    def tokenize(self, text: str,
+                 never_split: Optional[List[str]] = None) -> \
+            List[Optional[str]]:
+        r"""Basic Tokenization of a piece of text.
+
+        Split on "white spaces" only, for sub-word tokenization, see
+        WordPieceTokenizer.
+
+        Args:
+            never_split: (`optional`) list of str
+                Kept for backward compatibility purposes.
+                Now implemented directly at the base class level
+                List of token not to split.
+        """
+        never_split = self.never_split + (never_split
+                                          if never_split is not None else [])
+        text = self._clean_text(text)
+        # This was added on November 1st, 2018 for the multilingual and Chinese
+        # models. This is also applied to the English models now, but it doesn't
+        # matter since the English models were not trained on any Chinese data
+        # and generally don't have any Chinese data in them (there are Chinese
+        # characters in the vocabulary because Wikipedia does have some Chinese
+        # words in the English Wikipedia.).
+        if self.tokenize_chinese_chars:
+            text = self._tokenize_chinese_chars(text)
+        orig_tokens = whitespace_tokenize(text)
+        split_tokens = []
+        for token in orig_tokens:
+            assert token is not None
+            if self.do_lower_case and token not in never_split:
+                token = token.lower()
+                token = self._run_strip_accents(token)
+            split_tokens.extend(self._run_split_on_punc(token))
+
+        output_tokens = whitespace_tokenize(" ".join(split_tokens))
+        return output_tokens
+
+    @classmethod
+    def _run_strip_accents(cls, text: str) -> str:
+        r"""Strips accents from a piece of text."""
```

Comment:
To help undersand what the method really does, add a simple example of input & output in the docstring.